File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    big = []
    small = []
    # 각 쌍 중 큰거 한쪽으로 작은거 한쪽으로 몰아넣고
    for col, row in sizes:
        if col>row:
            big.append(col)
            small.append(row)
        else:
            big.append(row)
            small.append(col)   
    # 각 리스트에서 max값 추출해서 곱하기
    return max(big) * max(small)

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int big = 0, small = 0;
        // 배열 돌면서
        for(int[] size : sizes) {
            // 가로가 길다면
            if (size[0] > size[1]) {
                // big을 가장 긴걸로 초기화(가로)
                if (big < size[0]) big = size[0];
                // small을 가장 긴걸로 초기화(세로)
                if (small < size[1]) small = size[1];
            } else {
                // big을 가장 긴걸로 초기화(세로)
                if (big < size[1]) big = size[1];
                // small을 가장 긴걸로 초기화(가로)
                if (small < size[0]) small = size[0];
            }
        }
        return big*small;
    }
}